### PR TITLE
Capture Matching Context & Filter UI

### PR DESCRIPTION
**Pull request by bstetzer32**
## Description

Added a filter for the Capture Matching tool that is currently awaiting API support to be fully implemented. Context and provider for the Capture Matching tool was added to facilitate this ability, as well as refactoring elements from the CaptureMatchingView into their own components.

**Issue(s) addressed**

- Works Towards Resolving #326 

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [x] Refactor

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

The filter icon is just an icon that does nothing when pressed. All useEffects and variables related to the overall function of the Capture Matching Tool are located in the top level CaptureMatchingView.

**What is the new behavior?**

The filter icon opens up a side panel when pressed, allowing the user to enter a start and end date as well as organization. All useEffects and variables related to the overall function of the Capture Matching Tool have been moved to a CaptureMatchingContext that is provided at the CaptureMatchingView component, and the majority of the UI in the CaptureMatchingView has been moved to a CaptureMatchingTable component so that it can have access to the variables in the context.

## Breaking change

**Does this PR introduce a breaking change?**

No breaking changes.

## Other useful information

We are awaiting support for this filter in the [treetracker-api repository](https://github.com/Greenstand/treetracker-api) before this issue can be closed [(see treetracker-api issue 57)](https://github.com/Greenstand/treetracker-api/issues/57). When that support is added, changes will need to be made to the fetch functions in the context as well as their respective API functions, allowing them to have a filter passed into them.

### CODE REVIEW

File: src/components/CaptureMatching/CaptureMatchingFilter.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
+import React, { useContext } from 'react';
```
Comment (nmcharlton) [inline src/components/CaptureMatching/CaptureMatchingFilter.js:1]: This is fine to get us started, but I think we need to do some refinement of the UI.
I would expect the filter to appear as a drawer on top of everything else, similar to Capture Details or Grower Details, and the styling should be similar to the filter for Earnings and Payments.

This toggled style could work, but would need to be closer on the screen to the thing it's filtering.

File: src/components/CaptureMatching/CaptureMatchingFilter.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                <TextField
+                  data-testid="org-dropdown"
+                  select
+                  label="Organization"
+                  htmlFor="organization"
+                  id="organization"
+                  value={organizationId}
+                  onChange={(e) => setOrganizationId(e.target.value)}
+                >
+                  {[...defaultOrgList, ...orgList].map((org) => (
```
Comment (nmcharlton) [inline src/components/CaptureMatching/CaptureMatchingFilter.js:191]: I'm seeing a warning in the console:
> Material-UI: You have provided an out-of-range value `null` for the select component.
Consider providing a value that matches one of the available options or ''.
The available values are `ALL_ORGANIZATIONS`, `ORGANIZATION_NOT_SET`, `11`, `13`, `14`, `15`, `1`.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

**Comment by dadiorchen** [comment]
@bstetzer32 I am going to implement the filter based on some of your code, please let me know if there is any other consideration.

**Comment by bstetzer32** [comment]
@dadiorchen I have been neck deep in two different core base lately, so I can't think of any concerns off the top of my head. If you have any questions feel free to @ me.

**Comment by nmcharlton** [comment]
@dadiorchen Can this PR now be closed?

**Comment by dadiorchen** [comment]
Yes

**Pull request closed without merging**